From 1518a18f95df050e5f9aae570e7748e5fba3d01a Mon Sep 17 00:00:00 2001
From: Mattias Busck <mattias.busck@hostmobility.com>
Date: Mon, 26 Sep 2022 12:41:35 +0200
Subject: [PATCH] HMX: Modify customboard devicetree for HMX

Modify  arch/arm/dts/imx8mp-var-dart-dt8mcustomboard.dts to match hmx1
hardware.

* Console output is uart3
* Do not use i2c3
* Remove GPIO expander from customboard
---
 ...ar-dart-dt8mcustomboard-legacy-u-boot.dtsi |  4 +-
 ...mx8mp-var-dart-dt8mcustomboard-u-boot.dtsi |  4 +-
 .../dts/imx8mp-var-dart-dt8mcustomboard.dts   | 60 ++++---------------
 .../imx8mp_var_dart/imx8mp_var_dart.c         |  7 ++-
 include/configs/imx8mp_var_dart.h             |  4 +-
 5 files changed, 20 insertions(+), 59 deletions(-)

diff --git a/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-legacy-u-boot.dtsi b/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-legacy-u-boot.dtsi
index 727898df6a..8049ef565f 100644
--- a/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-legacy-u-boot.dtsi
+++ b/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-legacy-u-boot.dtsi
@@ -6,11 +6,11 @@
 
 #include "imx8mp-var-common-u-boot.dtsi"
 
-&uart1 {
+&uart3 {
 	u-boot,dm-spl;
 };
 
-&pinctrl_uart1 {
+&pinctrl_uart3 {
 	u-boot,dm-spl;
 };
 
diff --git a/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-u-boot.dtsi b/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-u-boot.dtsi
index 727898df6a..8049ef565f 100644
--- a/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-u-boot.dtsi
+++ b/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard-u-boot.dtsi
@@ -6,11 +6,11 @@
 
 #include "imx8mp-var-common-u-boot.dtsi"
 
-&uart1 {
+&uart3 {
 	u-boot,dm-spl;
 };
 
-&pinctrl_uart1 {
+&pinctrl_uart3 {
 	u-boot,dm-spl;
 };
 
diff --git a/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard.dts b/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard.dts
index 4e5ffcc90e..b9b378509c 100644
--- a/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard.dts
+++ b/arch/arm/dts/imx8mp-var-dart-dt8mcustomboard.dts
@@ -7,15 +7,10 @@
 #include "imx8mp-var-dart.dtsi"
 
 / {
-	model = "Variscite DART-MX8M-PLUS on DT8MCustomBoard 2.0 and above";
-
-	aliases {
-		gpio5 = &pca6408_1;
-		gpio6 = &pca6408_2;
-	};
+	model = "Host Mobility HMX1";
 
 	chosen {
-		stdout-path = &uart1;
+		stdout-path = &uart3;
 	};
 
 	reg_usdhc2_vmmc: regulator-usdhc2-vmmc {
@@ -32,9 +27,9 @@
 	};
 };
 
-&uart1 {
+&uart3 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_uart1>;
+	pinctrl-0 = <&pinctrl_uart3>;
 	status = "okay";
 };
 
@@ -46,16 +41,7 @@
 	scl-gpios = <&gpio5 16 GPIO_ACTIVE_HIGH>;
 	sda-gpios = <&gpio5 17 GPIO_ACTIVE_HIGH>;
 	status = "okay";
-};
-
-&i2c3 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default", "gpio";
-	pinctrl-0 = <&pinctrl_i2c3>;
-	pinctrl-1 = <&pinctrl_i2c3_gpio>;
-	scl-gpios = <&gpio5 18 GPIO_ACTIVE_HIGH>;
-	sda-gpios = <&gpio5 19 GPIO_ACTIVE_HIGH>;
-	status = "okay";
+	/* TODO gpio-expander, rtc, temp sensor, check Linux device tree */
 };
 
 &i2c4 {
@@ -66,27 +52,8 @@
 	scl-gpios = <&gpio5 20 GPIO_ACTIVE_HIGH>;
 	sda-gpios = <&gpio5 21 GPIO_ACTIVE_HIGH>;
 	status = "okay";
-
-	pca6408_1: gpio@20 {
-		compatible = "ti,tca6408";
-		reg = <0x20>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-
-	pca6408_2: gpio@21 {
-		compatible = "ti,tca6408";
-		reg = <0x21>;
-		gpio-controller;
-		#gpio-cells = <2>;
-
-		eth1_phy_rst_hog {
-			gpio-hog;
-			gpios = <0 0>;
-			output-high;
-			line-name = "eth1_phy_rst";
-		};
-	};
+	
+	/* TODO accelerometer, hsm, adapt from  Linux device tree*/
 };
 
 &usb3_phy0 {
@@ -166,10 +133,10 @@
 };
 
 &iomuxc {
-	pinctrl_uart1: uart1grp {
+	pinctrl_uart3: uart3grp {
 		fsl,pins = <
-			MX8MP_IOMUXC_UART1_RXD__UART1_DCE_RX				0x40
-			MX8MP_IOMUXC_UART1_TXD__UART1_DCE_TX				0x40
+			MX8MP_IOMUXC_UART3_RXD__UART3_DCE_RX				0x40
+			MX8MP_IOMUXC_UART3_TXD__UART3_DCE_TX				0x40
 		>;
 	};
 
@@ -180,13 +147,6 @@
 		>;
 	};
 
-	pinctrl_i2c3: i2c3grp {
-		fsl,pins = <
-			MX8MP_IOMUXC_I2C3_SCL__I2C3_SCL					0x400001c2
-			MX8MP_IOMUXC_I2C3_SDA__I2C3_SDA					0x400001c2
-		>;
-	};
-
 	pinctrl_i2c4: i2c4grp {
 		fsl,pins = <
 			MX8MP_IOMUXC_I2C4_SCL__I2C4_SCL					0x400001c2
diff --git a/board/variscite/imx8mp_var_dart/imx8mp_var_dart.c b/board/variscite/imx8mp_var_dart/imx8mp_var_dart.c
index 9de06292c3..e2df359753 100644
--- a/board/variscite/imx8mp_var_dart/imx8mp_var_dart.c
+++ b/board/variscite/imx8mp_var_dart/imx8mp_var_dart.c
@@ -36,8 +36,8 @@ DECLARE_GLOBAL_DATA_PTR;
 #define GPIO_PAD_CTRL	(PAD_CTL_DSE1 | PAD_CTL_PUE | PAD_CTL_PE  | PAD_CTL_HYS)
 
 static iomux_v3_cfg_t const uart_pads_dart[] = {
-	MX8MP_PAD_UART1_RXD__UART1_DCE_RX | MUX_PAD_CTRL(UART_PAD_CTRL),
-	MX8MP_PAD_UART1_TXD__UART1_DCE_TX | MUX_PAD_CTRL(UART_PAD_CTRL),
+	MX8MP_PAD_UART3_RXD__UART3_DCE_RX | MUX_PAD_CTRL(UART_PAD_CTRL),
+	MX8MP_PAD_UART3_TXD__UART3_DCE_TX | MUX_PAD_CTRL(UART_PAD_CTRL),
 };
 
 static iomux_v3_cfg_t const uart_pads_som[] = {
@@ -122,7 +122,8 @@ int board_early_init_f(void)
 	if (board_id == BOARD_ID_DART) {
 		imx_iomux_v3_setup_multiple_pads(uart_pads_dart,
 			ARRAY_SIZE(uart_pads_dart));
-		init_uart_clk(0);
+		/* Init clock for HMX on UART3, also in CONFIG_MXC_UART_BASE */
+		init_uart_clk(2);
 	}
 	else if (board_id == BOARD_ID_SOM) {
 		imx_iomux_v3_setup_multiple_pads(uart_pads_som,
diff --git a/include/configs/imx8mp_var_dart.h b/include/configs/imx8mp_var_dart.h
index 8ba09002a0..d2a3e43880 100644
--- a/include/configs/imx8mp_var_dart.h
+++ b/include/configs/imx8mp_var_dart.h
@@ -79,7 +79,7 @@
 	"image=Image.gz\0" \
 	"img_addr=0x42000000\0" \
 	"splashimage=0x50000000\0" \
-	"console=ttymxc0,115200\0" \
+	"console=ttymxc2,115200\0" \
 	"fdt_addr_r=0x43000000\0" \
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
@@ -214,7 +214,7 @@
 #define PHYS_SDRAM			0x40000000
 #define DEFAULT_SDRAM_SIZE		(512 * SZ_1M)
 
-#define CONFIG_MXC_UART_BASE		UART1_BASE_ADDR
+#define CONFIG_MXC_UART_BASE		UART3_BASE_ADDR
 
 /* Monitor Command Prompt */
 #define CONFIG_SYS_PROMPT_HUSH_PS2	"> "
-- 
2.30.2

